### DISCUSSION TITLE
Get charon daemon running configuration

### DISCUSSION THREAD

**Discussion by mrxanto** [strongswan/strongswan#2462] [Q&A] [answered]
Hello.

Is there a way to get all configuration from charon daemon?

It look like charon daemon (strongswan-charon - 5.9.14-2) on OpenWrt 23.05.4 won't read "retry_initiate_interval = 60s" variable from file /etc/strongswan.conf.

I've tried also dirty hacks with /etc/init.d/swanctl to wrote this option to /var/ipsec/strongswan.conf on start daemon stage,
but no luck - daemon receive "NO_PROPOSAL_CHOSEN" notify error (other side now isn't configured) and doesn't retry connection.

**Comment by tobiasbrunner** [comment, marked as answer]
> Is there a way to get all configuration from charon daemon?

No.

> but no luck - daemon receive "NO_PROPOSAL_CHOSEN" notify error (other side now isn't configured) and doesn't retry connection.

That option that not affect this. It only triggers a retry if the DNS resolution fails. Any fatal errors (like NO_PROPOSAL_CHOSEN or AUTHENTICATION_FAILED) will terminate the SA and there currently won't be a retry (use trap policies or manually re-trigger the initiation).

**Comment by mrxanto** [comment]
> It only triggers a retry if the DNS resolution fails

I've just misunderstood "e.g" abbreviation, thinking that these option are related not only DNS resolution fail and some other fatal errors too.

> use trap policies

Ok, now I try change start_action from "start" to "trap|start":

<details> 
  <summary># grep action /etc/swanctl/conf.d/tun.conf</summary>
<pre>
        start_action = trap|start
        dpd_action = trap
        close_action = trap
</pre>
</details>

Thanks.

**Comment by mrxanto** [comment]
> Any fatal errors (like NO_PROPOSAL_CHOSEN or AUTHENTICATION_FAILED) will terminate the SA and there currently won't be a retry (use trap policies

After some thought, I think that trap policy is not enough for some cases for recovery connection.
Let's assume that host with strongsWan (B) is behide NAT and will initiate connection to other host (A) with public IP address.
In the normal case traffic goes from local_ts host "A" to local_ts host "B". What will happen if host "A" loses connection with Internet?
Trap policy on host "B" not applicable, because nobody from "local_ts" host "B" will be try connect to "local_ts" host "A", and
host "A" can't re-establish connection to host "B", which is behide NAT.
Would in this case work option "retry_initiate_interval = 60s"? All host are defined with their IP addresses in configuration files.

**Reply by tobiasbrunner** [reply]
> Trap policy on host "B" not applicable, because nobody from "local_ts" host "B" will be try connect to "local_ts" host "A"

Why not? How was the connection initiated in the first place then? And if it doesn't happen organically, you have to generate traffic (e.g. by regularly pinging a remote host).

**Comment by mrxanto** [comment]
> Why not? 

Because hosts in "local_ts" host "A" poll hosts in "local_ts" host "B", not the other way around.

> How was the connection initiated in the first place then?

Connection are initiated by host "B" with:
<details> 
  <summary># grep action /etc/swanctl/conf.d/tun.conf</summary>
<pre>
        start_action = start
        dpd_action = restart
        close_action = start
</pre>
</details>

> And if it doesn't happen organically, you have to generate traffic (e.g. by regularly pinging a remote host)

Yes, ping some_from_local_ts_host_A -c 1 - probably will works, but ... it look as duct tape.
